### ISSUE TITLE
Adapt TaskPublish.py script to GO-based DBS

### ISSUE THREAD

**Issue by belforte** [dmwm/CRABServer#7350]
there are some small changes in API's syntax and return valued and some more relevant ones in semantic  of migration submission and queries, see new [DBS2GO documentation](https://github.com/dmwm/dbs2go/blob/master/docs/apis.md)  and full history in GH issues in here and in DBB2GO project and in https://github.com/dmwm/CRABServer/projects/8

I will create a TaskPublishGo.py and run it in parallel to current one publishing to testbed.

Let's keep here a list of things to do

- [x] modify PublisherMaster so that it calls the new script as well (but does not fail on it)
- [x] each API returns a list
- [x] submitMigration for one block will expand request to all ancestors as well, should use block_name, not migrationId to track completion
- [x] submitMigrations (all DBSApi calls) never return exception unless http connection to server fails. 
- [x] there's a new possible value for migration status : 4 migration request is already exist in DB, i.e. the requested block or dataset is already found in database
- [ ] can also try to request migration of full dataset "upfront". New server is expected to be able to deal with that.
- [x] write so that it can simply replace current one after validation
- [x] create independend OK/Fail log for the validation
- [ ] add POST via curl as fallback for block injections https://github.com/dmwm/CRABServer/issues/7314

**Comment by belforte** [comment]
now parasitically testing TaskPublishGo.py as per https://cms-logbook.cern.ch/elog/Analysis+Operations/3725
text pasted below:
```
I have added to Publisher code to perform migrations using new GO-based server and
have modified it so that in paralled to current publication to prod/phys03, it
also published user data to int/phys03 via cmsweb-testbed go-based servers.

Initially  I will do it for 10% of the tasks, since I expect more migrations to be needed
than for prod/phys03 and do not want to slow down too much the CRAB publication cycle.

I will be watching things and raising fraction in next days.
```

**Comment by belforte** [comment]
Note. Cannot publish datasets which have parents in `prod/phys03` due to 
https://github.com/dmwm/dbs2go/issues/69

- [ ] need to patch TaskPublishGo.py accordingly

**Comment by belforte** [comment]
Everything was done and tested with the exception of
 * full dataset migration : not needed, let's keep changes at minimum for the time being
 * fall back post via curl, now in HOLD, see  https://github.com/dmwm/CRABServer/issues/7314

**Comment by belforte** [comment]
- [ ] time to remove test /validation stuff from TaskPublishGo.py and make it ready for prime time

**Comment by belforte** [comment]
this is now completed. New version of TaskPublish is ready and I will add it to master repo in a new PR

**Comment by belforte** [comment]
TaskPublishGo.py added to master gia https://github.com/dmwm/CRABServer/pull/7373
